roles/get_logs_from_namespace/tasks/main.yml

```
@@ -10,9 +10,13 @@
     path: "{{ glfn_dir }}"
   register: dir_stat
   when: glfn_dir is defined
-  failed_when:
-    - glfn_dir is defined
-    - not dir_stat.stat.exists
+
+- name: Create the logs directory if defined but does not exist
+  ansible.builtin.file:
+    path: "{{ glfn_dir }}"
+    state: directory
+    mode: '0775'
+  when: glfn_dir is defined and (dir_stat is not defined or not dir_stat.stat.exists)
```

Avoid the stat of the `glfn_dir` by always creating the `glfn_dir` (since the task is idempotent), simplifying the conditional:

```suggestion
  when: glfn_dir is defined
```

---

merged the suggestion, thank you!